astropy/io/fits/hdu/hdulist.py

```diff
@@ -787,6 +782,22 @@ def filename(self):
         return None
 
     @classmethod
+    def _ext_equal(self, ext, hdu, idx):
```

Comment:
This doesn't use `self` (and for `classmethods` the "self" is typically spelled "cls").  But in any case this can be a `staticmethod` instead.  I'd also prefer a different name...maybe something like "_ext_match".